lib/hash_police/police.rb

```diff
@@ -13,7 +13,11 @@ def initialize rule, context_key = ""
     def check target
       result = CheckResult.new(context_key)
```

Comment:
Here I think we can separate the `proc` and `type` matching, since "matching by proc" is not the same as "matching by type" semantically.

```ruby
if rule.kind_of?(Proc) && ! rule.call(target)
  result.invalid_by_proc
  return result
end

unless type_matched?(rule_target)
   ...
end
```